File: advertising_sdk/src/main/java/com/adadapted/android/sdk/core/session/SessionAdapter.java
package com.adadapted.android.sdk.core.session;

import com.adadapted.android.sdk.core.device.DeviceInfo;

public interface SessionAdapter {
    interface SessionInitListener {
        void onSessionInitialized(Session session);
        void onSessionInitializeFailed();
    }

    interface AdGetListener {
        void onNewAdsLoaded(Session session);
        void onNewAdsLoadFailed();
    }

    interface Listener extends SessionInitListener, AdGetListener {}

    void sendInit(DeviceInfo deviceInfo, SessionInitListener listener);

    void sendRefreshAds(Session session, AdGetListener listener);
}

File: advertising_sdk/src/main/java/com/adadapted/android/sdk/core/session/SessionAdapter.kt
package com.adadapted.android.sdk.core.session

import com.adadapted.android.sdk.core.device.DeviceInfo

interface SessionAdapter {
    interface SessionInitListener {
        fun onSessionInitialized(session: Session)
        fun onSessionInitializeFailed()
    }

    interface AdGetListener {
        fun onNewAdsLoaded(session: Session)
        fun onNewAdsLoadFailed()
    }

    interface Listener : SessionInitListener, AdGetListener

    fun sendInit(deviceInfo: DeviceInfo, listener: SessionInitListener)
    fun sendRefreshAds(session: Session, listener: AdGetListener)
}